### ISSUE TITLE
Pruning for Seraphis transactions?

### ISSUE THREAD

**Issue by rbrunner7** [seraphis-migration/wallet3#9]
There is a mechanism called *pruning* where the blockchain stores the full information only about 1/8th of all transactions; the information stored about the other 7/8th is "pruned" i.e. reduced to a minimum that is needed for the daemon and wallets to still use the transactions. This reduces the size of the blockchain file substantially, to about 1/3rd, to 50 GB instead of 150 GB at the time of writing.

Read more general info about pruning e.g. in this [getmonero.org blog post](https://www.getmonero.org/resources/moneropedia/pruning.html).

If you go through the Monero code base you frequently stumble over bits and pieces of code that deals with pruning, with many alternative code paths for pruned instead of "full" transactions. [moneromooo's PR](https://github.com/monero-project/monero/pull/4843) with the original implementation changes no less than 52 files.

This all leads to the question of this issue: Will there pruning for Seraphis transactions? And if yes, what does that mean for the implementation of the wallet?

**Comment by UkoeHB** [comment]
Yes I think Seraphis txs should be prunable. It should not matter for seraphis wallets, which should only care about managing the user's transaction activity (history of created txs, balance recovery, new transactions, address book [I kind of wonder if this even needs to be in the domain of wallets aside from maybe read-only hmm]).
